service-map/src/main/java/fi/mml/map/mapwindow/util/OskariLayerWorker.java

```diff
@@ -145,7 +204,103 @@ public static JSONObject getListOfMapLayers(final List<OskariLayer> layers, fina
                     else {
                         FORMATTER.removeAdminInfo(layerJson);
                     }
+                    //log.debug("Adding layer to list");
+                    layersList.put(layerJson);
+                }
+            }
+            catch(Exception ex) {
+                log.error(ex);
+            }
+        }
+        log.debug("Created JSON in", System.currentTimeMillis() - start, "ms");
+        log.debug("Returning", layersList.length(), "/", layers.size(),"layers");
+
+        final JSONObject result = new JSONObject();
+        JSONHelper.putValue(result, KEY_LAYERS, layersList);
+        return result;
+    }
+    
+    /**
+     * Gets all the map layers the user is permitted to view
+     *
+     * @param layers      list of layers to render
+     * @param user        User
+     * @param lang        Language
+     * @param isPublished Determines the permission type used for the layers (view/published view)
+     * @param isSecure    true to modify urls for easier proxy forwarding/false to keep as is
+     * @return JSONObject of map layers
+     */
+    public static JSONObject getListOfMapLayers(final List<OskariLayer> layers, final User user,
+                                                final String lang, final boolean isPublished, final boolean isSecure) {
```

Comment:
Is this a copy of the method above, but without the crs param? Just call the above method instead and handle null crs instead.

Reply:
So instead of copy/pasting the implementation call getListOfMapLayers(layers, user, lang, null, isPublished, isSecure) implemented above.